llvm/lib/Target/AMDGPU/AMDGPUResourceUsageAnalysis.cpp

```diff
@@ -239,6 +247,15 @@ AMDGPUResourceUsageAnalysis::analyzeResourceUsage(
           break;
         }
 
+        // For functions that use the llvm.amdgcn.init.whole.wave intrinsic, we
+        // often add artificial VGPR arguments for the purpose of preserving
+        // their inactive lanes. These should not be reported as part of our
+        // VGPR usage. We can identify them easily because they're only used in
+        // the chain call, and possibly in an IMPLICIT_DEF coming from an
+        // llvm.amdgcn.dead intrinsic.
+        if (IsIWWFunction && IsChainCall && TRI.isVectorRegister(MRI, Reg))
+          continue;
```

Comment:
extend to any implicit argument that isn't in the MCInstrDesc, ignoring variadic instructions in the variadic section